src/snowflake/cli/plugins/nativeapp/codegen/snowpark/callback_source.py.jinja

```diff
@@ -12,107 +14,155 @@ except ModuleNotFoundError as exc:
     )
     sys.exit(1)
 
-orig_globals = globals().copy()
-
-found_correct_version = hasattr(
+__snowflake_internal_found_correct_version = hasattr(
     snowflake.snowpark.context, "_is_execution_environment_sandboxed_for_client"
 ) and hasattr(snowflake.snowpark.context, "_should_continue_registration")
 
-if not found_correct_version:
+if not __snowflake_internal_found_correct_version:
     print(
         "Did not find the minimum required version for snowflake-snowpark-python package. Please upgrade to v1.15.0 or higher.",
         file=sys.stderr,
     )
     sys.exit(1)
 
-__snowflake_cli_native_app_internal_callback_return_list: List[Any] = []
+__snowflake_global_collected_extension_fn_json = []
+
+def __snowflake_internal_create_extension_fn_registration_callback():
+    def __snowflake_internal_try_extract_lineno(extension_fn):
+        try:
+            return inspect.getsourcelines(extension_fn)[1]
+        except Exception:
+            return None
+
+    def __snowflake_internal_extract_extension_fn_name(extension_fn):
+        try:
+            import snowflake.snowpark._internal.utils as snowpark_utils
+
+            if hasattr(snowpark_utils, 'TEMP_OBJECT_NAME_PREFIX'):
```

Comment:
Is this a version guard? Is it necessary given the v1.15.0 check above and the exception wrapped around?

Reply:
No, it's pretty much "undoing" a bad transformation, but pulling the authority from Snowpark's internal APIs. Even with the version check passing, they could still rename / remove this field, and we'd fall back to the generic logic. Not ideal, but we have a channel open with Snowpark to push improvements upon this initial solution. That is one IMHO.